src/assets/wise5/authoringTool/peer-grouping/edit-peer-grouping-dialog/edit-peer-grouping-dialog.component.ts

```diff
@@ -25,14 +31,63 @@ export class EditPeerGroupingDialogComponent extends AuthorPeerGroupingDialogCom
   ngOnInit(): void {
     this.peerGrouping = this.utilService.makeCopyOfJSONObject(this.peerGrouping);
     this.stepsUsedIn = this.peerGroupingAuthoringService.getStepsUsedIn(this.peerGrouping.tag);
+    this.logicType = this.getLogicType(this.peerGrouping.logic);
+    if (this.logicType === DIFFERENT_IDEAS_VALUE) {
+      this.referenceComponent = this.getDifferentIdeasReferenceComponent(this.peerGrouping.logic);
+    }
+  }
+
+  private getLogicType(logic: string): string {
+    if (new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic) != null) {
+      return DIFFERENT_IDEAS_VALUE;
+    } else {
+      return logic;
+    }
+  }
+
+  private getDifferentIdeasReferenceComponent(logic: string): any {
+    const result = new RegExp(DIFFERENT_IDEAS_REGEX).exec(logic);
+    return {
+      nodeId: result[1],
+      componentId: result[2]
+    };
+  }
+
+  referenceComponentNodeIdChanged(event: any): void {
```

Comment:
This function is very similar to EditDynamicPromptComponent.referenceComponentNodeIdChanged(). Can we extract to a common parent (abstract?) class?

Reply:
We can do this in another issue. Should it be a common parent class or a utility function? I think we should rename \<edit-connected-component-node-select\> and \<edit-connected-component-component-select\> now that we use them in more places besides connected component authoring. Then we can refactor these common functions too.

edit-connected-component.ts => automaticallySetConnectedComponentComponentIdIfPossible()
edit-peer-grouping-dialog.ts => referenceComponentNodeIdChanged()
edit-dynamic-prompt.component.ts => referenceComponentNodeIdChanged()

Reply:
Ok. I think a common abstract parent class would be good. I'll create a new issue for it.